### DISCUSSION TITLE
Stream per-kernel statistics

### DISCUSSION THREAD

**Discussion by Sacusa** [accel-sim/accel-sim-framework#190] [Q&A] [answered]
I want to run concurrent kernels using CUDA Streams and get statistics for each kernel _individually_. However, it looks like GPGPU-Sim runs all the kernels that can run together (i.e., kernels at the head of their respective streams) and prints out statistics for all of them together. Is it possible to get per-kernel statistics?

**Comment by JRPan** [comment, marked as answer]
We are working on that right now actually. 
Please check 
https://github.com/accel-sim/accel-sim-framework/pull/166
and
https://github.com/accel-sim/gpgpu-sim_distribution/pull/47

Theses will be merged to [stream-stats](https://github.com/accel-sim/gpgpu-sim_distribution/tree/stream-stats) soon. Then will be merged into dev and then release eventually. 

The stats are per stream on that branch.
